internal/common/metrics.go

```diff
@@ -53,6 +53,28 @@ func sendCollectDMetrics(metricPerMinute int, duration time.Duration) error {
 	defer ticker.Stop()
 	endTimeout := time.After(duration)
 
+	// Sending the collectd metric within the first minute before the ticker kicks in the next minute
+	for t := 0; t < metricPerMinute; t++ {
+		err = client.Write(ctx, &api.ValueList{
+			Identifier: api.Identifier{
+				Host:   exec.Hostname(),
+				Plugin: fmt.Sprint(t),
+				Type:   "gauge",
+			},
+			Time:     time.Now(),
+			Interval: time.Minute,
+			Values:   []api.Value{api.Gauge(t)},
+		})
```

Comment:
```
&api.ValueList{
			Identifier: api.Identifier{
				Host:   exec.Hostname(),
				Plugin: fmt.Sprint(t),
				Type:   "gauge",
			},
			Time:     time.Now(),
			Interval: time.Minute,
			Values:   []api.Value{api.Gauge(t)},
		}
```
looks duplicated in places.

I suggest a helper/constructor to return the `api.ValueList`.

```
func newValueList(val int) *api.ValueList {
...
}